uvm/bootstrap/com/untangle/uvm/UvmContextSelector.java

```diff
@@ -0,0 +1,248 @@
+/**
+ * $Id$
+ */
+
+package com.untangle.uvm;
+
+import org.apache.logging.log4j.core.Appender;
+import org.apache.logging.log4j.core.LoggerContext;
+import org.apache.logging.log4j.core.appender.SyslogAppender;
+import org.apache.logging.log4j.core.config.Configuration;
+import org.apache.logging.log4j.core.config.ConfigurationSource;
+import org.apache.logging.log4j.core.config.LoggerConfig;
+import org.apache.logging.log4j.core.config.xml.XmlConfiguration;
+import org.apache.logging.log4j.core.layout.PatternLayout;
+import org.apache.logging.log4j.core.net.Facility;
+import org.apache.logging.log4j.core.net.Protocol;
+import org.apache.logging.log4j.core.selector.ContextSelector;
+import org.apache.logging.log4j.status.StatusLogger;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Custom ContextSelector implementation to create multiple logger context for apps
+ */
+public class UvmContextSelector implements ContextSelector {
+
+    private final ConcurrentHashMap<String, LoggerContext> contexts = new ConcurrentHashMap<>();
```

Comment:
It is advised to have the reference variable of type interface `ConcurrentMap<String, LoggerContext>`

Reply:
Changed to `ConcurrentMap<String, LoggerContext>`